### ISSUE TITLE
Unstable table grid line width with 125% scale on Windows

### ISSUE THREAD

**Issue by electriq** [JFormDesigner/FlatLaf#152]
Hi! I found that if the scale ('font size') is set to 125% (common on 15" FullHD laptop; Windows 10), the table grid is drawn with 1 or 2 pixel line depending on current position of the table inside the scroll pane. Sometomes some lines are 1 px and some 2 px. It also switching haotically while moving an internal frame with a table in it.

The only difference between these two screenshots is scrolling the table down. The horizintal lines become 2 px from 1 px. Vertical line are 1 and 2 px:

![Table1](https://user-images.githubusercontent.com/642565/89341408-f1a28b00-d6a9-11ea-9321-66e3c675de84.png)
![Table2](https://user-images.githubusercontent.com/642565/89341412-f2d3b800-d6a9-11ea-8a13-435cbf746ee8.png)

**Comment by DevCharly** [comment]
Yes, this is a common problem with painting in Java on scaled displays.
You never know whether a line becomes 1px or 2px thick (without anti-aliasing).

Window LaF and Metal have the same problems.
On the other hand, Nimbus and Substance do not have this problem.

FlatLaf currently uses `BasicTableUI` for painting the table. Unfortunately all painting methods in `BasicTableUI` are private and it is not possible to change/improve something without re-implementing the whole table painting in `FlatTableUI`.

I tried to use anti-aliasing, which looks better, but the grid lines are too thick IMHO (125%; Java 14):

![image](https://user-images.githubusercontent.com/5604048/89679158-dd040400-d8f0-11ea-844a-0ce996383119.png)

And with anti-aliasing the selection has gaps if grid lines are disabled, which is the default:

![image](https://user-images.githubusercontent.com/5604048/89679376-3ec46e00-d8f1-11ea-9938-f9dc3b4f349a.png)

Grid lines are disabled, but anti-aliasing causes gaps between cells:

![image](https://user-images.githubusercontent.com/5604048/89680610-8f3ccb00-d8f3-11ea-9cde-f90e4fa76a8f.png)

Probably have to re-implement table painting in `FlatTableUI` to fix this.
But this takes time...

For now I would recommend to use a lighter color for the grid lines. Then it is harder to notice 😉 

If you want try anti-aliasing yourself, add following to `FlatTableUI`:
~~~java
@Override
public void paint( Graphics g, JComponent c ) {
	FlatUIUtils.setRenderingHints( (Graphics2D) g );
	super.paint( g, c );
}
~~~

**Comment by DevCharly** [comment]
Fixed in commit 5ef0c9aae14348f52be047e837aa9ac856fe5a10

It turned out that the fix for https://bugs.openjdk.java.net/browse/JDK-8133919
in commit https://github.com/openjdk/jdk/commit/803735d54775b4ec5b7a4b649d99cc5f0b21f71f
is the reason for the unstable grid line width. (since Java 9)

`BasicTableUI.paintGrid()` now uses `Graphics.fillRect()` instead of `drawLine()` for painting grid lines.

Nimbus does not have unstable grid line width because it has a copy of `paintGrid()`, which still uses `drawLine()`.

**Comment by electriq** [comment]
With this patch (I did't tested before your release, sorry) I see unstable grid over the selection (background/grid color). And we have lost the right border of the custom row header which is a simple one-column JTable which does not draw the last vertical line now (maybe this is not a bug if last vertical line is not specifyed in docs, I don't know -- it just worked before).

![2020-12-05_203410](https://user-images.githubusercontent.com/642565/101259319-411cc480-3739-11eb-8758-a34860933425.png) ![2020-12-05_203136](https://user-images.githubusercontent.com/642565/101259278-fe5aec80-3738-11eb-8487-f5e374a03d64.png)

**Comment by DevCharly** [comment]
Ah, now I see it too. That's probably because the lines are now thinner than the space between the cells...
Need to fix cell bounds too. Have to investigate whether this is possible...

The fix for #46 is the reason for the missing right border of the row header...

**Comment by electriq** [comment]
OK, fixing our row header border was simple :)

```
    rowHeader = new JTable(rowTableModel)
    {
      @Override
      protected void paintComponent(Graphics g)
      {
        super.paintComponent(g);
        
        g.setColor(table.getGridColor());
        g.drawLine(getWidth() - 1, 0, getWidth() - 1, getHeight());
      }
    };
```

It seems you and us will end up with paintComponent() 'overloading' every pixel of Swing's default UI :)))
